trieste/acquisition/rule.py

```diff
@@ -425,36 +398,39 @@ def acquire(
 
         y_min = tf.reduce_min(dataset.observations, axis=0)
 
-        if state is None:
-            eps = 0.5 * (global_upper - global_lower) / (5.0 ** (1.0 / global_lower.shape[-1]))
-            is_global = True
-        else:
-            tr_volume = tf.reduce_prod(
-                state.acquisition_space.upper - state.acquisition_space.lower
-            )
-            step_is_success = y_min < state.y_min - self._kappa * tr_volume
-
-            eps = (
-                state.eps
-                if state.is_global
-                else state.eps / self._beta
-                if step_is_success
-                else state.eps * self._beta
-            )
+        def go(state: TrustRegion.State | None) -> tuple[TrustRegion.State | None, TensorType]:
```

Comment:
the type checking on these is horrible. Because `tf.Tensor` is untyped both as an object and a class, if you get the return values the wrong way round, mypy doesn't notice, even if your state is typed